src/kbmod/masking.py

```diff
@@ -2,6 +2,44 @@
 """
 
 import kbmod.search as kb
+from kbmod.search import RawImage
+
+
+def mask_trajectory(trj, stack, r, bitmask=1):
+    """Will apply a circular mask of radius `r` around
+    the trajectory's predicted position in each image.
+
+    Attributes
+    ----------
+    trj : `Trajectory`
+        The trajectory along which to apply the mask.
+    stack : `ImageStack`
+        The stack of images to apply the mask to.
+    r : `float`
+        The radius of the circular mask to apply.
+
+    Returns
+    -------
+    stack : `ImageStack`
+        The stack after the masks have been applied.
+    """
+
+    width = stack.get_width()
+    height = stack.get_height()
+
+    for i in range(stack.img_count()):
+        time = stack.get_single_image(i).get_obstime() - stack.get_single_image(0).get_obstime()
+
+        origin_of_mask = (trj.get_x_pos(time), trj.get_y_pos(time))
+        trajectory_mask = RawImage(width, height)
+
+        for y in range(height):
+            for x in range(width):
+                if (x - origin_of_mask[0]) ** 2 + (y - origin_of_mask[1]) ** 2 <= r**2:
```

Comment:
In general, I think we are going to have `r` much smaller than `width` or `height`. So it might be more efficient to do something like:

```
for dy in range(-r, r+1):
    for dx in range(-r, r+1):
        if dx**2 + dy**2 <= r**2:
            x = origin_of_mask[0] + dx
            y = origin_of_mask[1] + dy
            if (x >= 0) and (x < width) and (y >= 0) and (y < height):
...
```

This is what I was thinking when I made the comment about bounds checking.

Reply:
This is much better. Switched to this